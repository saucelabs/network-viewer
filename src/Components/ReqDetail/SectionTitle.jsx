import React from 'react';
import PropTypes from 'prop-types';

import { SECTION_TITLES, PAYLOAD_CAPTIONS } from '../../constants';
import Styles from './ReqDetail.styles.scss';
import IconCaretDown from '../../icons/IconCaretDown';
import IconCaretRight from '../../icons/IconCaretRight';

const SectionTitle = ({
  onClick,
  eventKey,
  isEncodeEnabled,
  isOpen,
  onPayloadTransform,
  isPayloadTransformed,
  isParseEnabled,
}) => {
  const payloadStatus = PAYLOAD_CAPTIONS[isParseEnabled ? 'parse' : 'encode'][isPayloadTransformed];

  return (
    <div className={Styles['section-title-row']}>
      <span
        className={Styles['section-title']}
        onClick={() => onClick(SECTION_TITLES[eventKey].key)}
        role="button"
        tabIndex={0}
      >
        {isOpen ?
          <IconCaretDown className={Styles['caret-icon']} /> :
          <IconCaretRight className={Styles['caret-icon']} />}
        {SECTION_TITLES[eventKey].name}
      </span>
      {(isEncodeEnabled || isParseEnabled) && (
        <span
          className={Styles['encode-url']}
          onClick={onPayloadTransform}
          role="button"
          tabIndex={0}
        >
          {`view ${payloadStatus}`}
        </span>
      )}
    </div>
  );
};

SectionTitle.propTypes = {
  eventKey: PropTypes.string.isRequired,
  isEncodeEnabled: PropTypes.bool,
  isOpen: PropTypes.bool,
  isParseEnabled: PropTypes.bool,
  isPayloadTransformed: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  onPayloadTransform: PropTypes.func.isRequired,
};

SectionTitle.defaultProps = {
  isEncodeEnabled: false,
  isOpen: false,
  isParseEnabled: false,
  isPayloadTransformed: true,
};

export default SectionTitle;
