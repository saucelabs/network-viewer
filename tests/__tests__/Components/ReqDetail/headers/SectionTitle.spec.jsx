import React from 'react';
import { shallow } from 'enzyme';

import SectionTitle from '../../../../../src/Components/ReqDetail/SectionTitle';

describe('SectionTitle', () => {
  const props = {
    eventKey: 'general',
    onClick: () => {},
    onPayloadTransform: () => {},
  };

  it('renders without crashing', () => {
    const element = shallow(<SectionTitle {...props} />);
    expect(element).toMatchSnapshot();
  });
});
