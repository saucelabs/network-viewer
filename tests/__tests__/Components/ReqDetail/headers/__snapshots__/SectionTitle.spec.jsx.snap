// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTitle renders without crashing 1`] = `
<div>
  <span
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <IconCaretUp
      className=""
    />
    General
  </span>
</div>
`;
